modules/core/arrow-core/src/main/kotlin/arrow/core/Either.kt

```diff
@@ -266,6 +266,28 @@ inline fun <A, B> EitherOf<A, B>.getOrHandle(default: (A) -> B): B =
 inline fun <A, B> EitherOf<A, B>.filterOrElse(predicate: (B) -> Boolean, default: () -> A): Either<A, B> =
   flatMap { if (predicate(it)) Right(it) else Left(default()) }
 
+/**
+ * * Returns [Either.Right] with the existing value of [Either.Right] if this is a [Either.Right] and the given
+ * predicate holds for the right value.
+ * * Returns `Left(default({right}))` if this is a [Either.Right] and the given predicate does not
+ * hold for the right value. Useful for error handling where 'default' returns a message with context on why the value
+ * did not pass the filter
+ * * Returns [Either.Left] with the existing value of [Either.Left] if this is a [Either.Left].
+ *
+ * Example:
+ * {: data-executable='true'}
+ * ```kotlin:ank
+ *
+ * Right(12).filterOrOther({ it > 10 }, { -1 }) // Result: Right(12)
+ * Right(7).filterOrOther({ it > 10 }, { "Value '$it' not greater than 10" })  // Result: Left("Value '7' not greater than 10")
+ *
+ * val left: Either<Int, Int> = Left(12)
+ * left.filterOrOther({ it > 10 }, { -1 })      // Result: Left(12)
+ * ```
+ */
```

Comment:
There are a couple of issues with the docs we've not flagged, let me give you the comment to have:

```
/**
 * * Returns [Either.Right] with the existing value of [Either.Right] if this is a [Either.Right] and the given
 * predicate holds for the right value.
 * * Returns `Left(default({right}))` if this is a [Either.Right] and the given predicate does not
 * hold for the right value. Useful for error handling where 'default' returns a message with context on why the value
 * did not pass the filter
 * * Returns [Either.Left] with the existing value of [Either.Left] if this is a [Either.Left].
 *
 * Example:
 *
 * {: data-executable='true'}
 * ```kotlin:ank
 * Right(12).filterOrOther({ it > 10 }, { -1 }) // Result: Right(12)
 * Right(7).filterOrOther({ it > 10 }, { "Value '$it' not greater than 10" }) 
 * ```
 *
 * {: data-executable='true'}
 * ```kotlin:ank
 * val left: Either<Int, Int> = Left(12)
 * left.filterOrOther({ it > 10 }, { -1 })
 * ```
 */
```

Reply:
Ok, this is beginning to make sense. So if there can only be one result/block, I assume that the two `Right` examples need to be split also?

I'll assume that's correct, and update the PR accordingly.

Thank you

Reply:
Correct, I missed it!